shell/platform/android/io/flutter/embedding/android/AndroidTouchProcessor.java

```diff
@@ -264,7 +265,8 @@ private void addPointerForIndex(
     packet.putLong(motionEventId); // motionEventId
     packet.putLong(timeStamp); // time_stamp
     if (isTrackpadPan) {
-      packet.putLong(getPointerChangeForPanZoom(pointerChange)); // change
+      panZoomType = getPointerChangeForPanZoom(pointerChange);
+      packet.putLong(panZoomType); // change
```

Comment:
What happens if we have -1 here? I think it will just cause a later crash. Maybe you can add a test case for some new unhandled type of event (future-proofing). And make sure we can handle/skip the event without a crash.

Reply:
No,I don't think so. The only leak case here is ACTION_SCROLL. It's mapped to PointerChange.HOVER and not handled by getPointerChangeForPanZoom(). For other case it will be handled by checking pointerChange value.
If it's -1, addPointerForIndex will drop this event